### ISSUE TITLE
Netflow v5 -- Unknown

### ISSUE THREAD

**Issue by mugugnu** [cloudflare/goflow#89]
Hi,
my goflow container (latest 3.4.3)  is receiving flow records with initial flags like this: (i'm inspecting them via nfcapd on a VM)

Flow Record:
  Flags        =              0x00 NETFLOW v5, Unsampled
  label        =            <none>

However goflow returns continously this error:
Error from: NetFlow (0) duration: 12.92µs. Unknown NetFlow version 5 (only decodes v9 and v10/IPFIX)"

Does goflow support netflow v5 ? Can you help me troubleshoot this ?

**Comment by lspgn** [comment]
Hello,
You need to send NetFlow v5 on a different port than v9/ipfix (by default :2056).
Let me know if that works.

**Comment by mugugnu** [comment]
I was actually sending to 2056, but goflow default setup was and still is:
time="2021-01-08T15:30:10Z" level=info msg="Listening on UDP :6343" Type=sFlow
time="2021-01-08T15:30:10Z" level=info msg="Listening on UDP :2055" Type=NetFlow
time="2021-01-08T15:30:10Z" level=info msg="Listening on UDP :2056" Type=NetFlowLegacy

I'm not sure what i actually did except for deleting the pods (both goflow and nflow-gen) and recreating them a bunch of times.

As long as goflow was configured to send to kafka i could not see any message in the goflow logs, when i addeded the -kafka=false i started seeing netflow traffic arrive.

So, bottom line is that i'm sending to 2056...logs read "netflowlegacy", sounds like v5 doesn't it ?

**Comment by lspgn** [comment]
> As long as goflow was configured to send to kafka i could not see any message in the goflow logs, when i addeded the -kafka=false i started seeing netflow traffic arrive.

Correct, it's only logging when Kafka is disabled.

> So, bottom line is that i'm sending to 2056...logs read "netflowlegacy", sounds like v5 doesn't it ?

Correct. I use `legacy` term for v5.

Do you still see the error about the version? If so, could you send me a pcap?

**Comment by mugugnu** [comment]
Without kafka it's working fine, no more errors.

**Comment by lspgn** [comment]
>  Update: also with kafka enabled, i'm receiving stuff on the choosen topic. I just don't know how to decode it (is it binary ?)

It is a [protobuf message](https://developers.google.com/protocol-buffers). You need to unmarshall the data.

**Comment by mugugnu** [comment]
everything's fine. Thanks for your support

**Comment by StefanoKoo** [comment]
@mugugnu Did you find a way to decode the messages?
